src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,114 @@
+        /// <see cref="OnnxScoringEstimator"/>s with specific models included in a package together with that extension method.
+        /// For an example, see Microsoft.ML.DnnImageFeaturizer.ResNet18 </param>
+        /// <param name="input">Input column name.</param>
+        /// <param name="output">Output column name.</param>
+        public DnnImageFeaturizerEstimator(IHostEnvironment env, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> modelFactory, string input, string output)
+        {
+            _modelChain = modelFactory(new DnnImageModelSelector(), env, input, output);
+        }
+
+        /// <summary>
+        /// Note that OnnxEstimator which this is based on is a trivial estimator, so this does not do any actual training,
+        /// just verifies the schema.
+        /// </summary>
+        public TransformerChain<OnnxTransform> Fit(IDataView input)
+        {
+            return _modelChain.Fit(input);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            return _modelChain.GetOutputSchema(inputSchema);
+        }
+    }
+
+    public static class DnnImageFeaturizerStaticExtensions
+    {
+        private sealed class OutColumn : Vector<float>
+        {
+            public PipelineColumn Input { get; }
+
+            public OutColumn(Vector<float> input, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> modelFactory)
+                : base(new Reconciler(modelFactory), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> _modelFactory;
+
+            public Reconciler(Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> modelFactory)
+            {
+                _modelFactory = modelFactory;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var outCol = (OutColumn)toOutput[0];
+                return new DnnImageFeaturizerEstimator(env, _modelFactory, inputNames[outCol.Input], outputNames[outCol]);
+            }
+        }
+
+        /// <summary>
+        /// Creates a DnnImageFeaturizer transform to be used by the static API.
+        /// <see cref="DnnImageFeaturizerEstimator"/> for more information about how the transformation works.
```

Comment:
You said that "Create a Dnn*" but the thing actually gets returned is a `OutColumn`.